File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
import sys

n = int(sys.stdin.readline().rstrip())
words = []

for _ in range(n):
    words.append(sys.stdin.readline().rstrip())

words = list(set(words))

# 사전 순 정렬
words.sort()

# 길이 순 정렬
words.sort(key=lambda x: len(x))

for word in words:
    print(word)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.cc
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <set>

using namespace std;

bool compare(string a, string b) {
    if (a.length() != b.length()) {
        return a.length() < b.length();
    }

    return a < b;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n;
    cin >> n;

    set<string> words;

    for (int i = 0; i < n; i++) {
        string word;
        cin >> word;
        words.insert(word);
    }

    vector<string> v(n);

    copy(words.begin(), words.end(), v.begin());
    sort(v.begin(), v.end(), compare);

    for (int i = 0; i < n; i++) {
        if (v[i] != "") {
            cout << v[i] << "\n";
        }
    }

    return 0;
}
